controllers/nodemaintenance_controller.go

```diff
@@ -91,10 +91,11 @@ type NodeMaintenanceReconciler struct {
 func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.logger = log.FromContext(ctx)
 	r.logger.Info("Reconciling NodeMaintenance")
+	defer r.logger.Info("Reconcile completed")
 
-	// Fetch the NodeMaintenance instance
-	instance := &nodemaintenancev1beta1.NodeMaintenance{}
-	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	// Fetch the NodeMaintenance instance (nm)
+	nm := &nodemaintenancev1beta1.NodeMaintenance{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, nm)
```

Comment:
something for another PR probably: use the given context

Reply:
Or now :)